### PR TITLE
[MT-68] Add ControlMaster ssh options in `mila init`

### PR DESCRIPTION
**Pull request by lebrice**
- Adds the `ControlMaster`, `ControlPath` and `ControlPersist` options to the `mila` ssh configuration entry as well as the one used to connect to compute nodes.
- Updates the entries in-place if they already exist.

These options make it possible to reuse an existing ssh connection via what's called "SSH multiplexing". I'm not an expert, but basically, it seems to create a file in a predetermined path (set via `"ControlPath"`) for each "real" ssh connection. Then, if you try to make a new SSH connection to the same host and port, that initial connection is reused instead of making a new one.

TL;DR: This makes it a LOT quicker to connect to the cluster or to compute nodes: after the first connection to the login node or compute node is done (still takes a while), the following connections are near-instant.

ALSO:
- Removes extra newlines between SSH configuration entries
    - This currently has a potentially annoying side-effect of collapsing existing empty block of > 2 empty lines in the user's ~/.ssh/config file into just a single empty line. 
- Improves the name and format of regression files used in unit tests. They are now readable Markdown files in this format:

    
    ```
    Running the `mila init` command with this initial content:
    
    (the initial content)
    
    and these use inputs: (the user inputs)
    leads to the following ssh config contents:
    
    (the resulting content of ~/.ssh/config)
    ```
    
    
(See https://github.com/lebrice/milatools/blob/ControlMaster-mila-init-MT-68/tests/cli/test_init_command/test_with_existing_entries_has_mila_compute_entry_has_mila_cpu_entry_no_mila_entry_.md for an example)

### CODE REVIEW

File: milatools/cli/utils.py [hunk 1]
```diff
@@ -139,14 +139,15 @@ def shjoin(split_command):
 class SSHConfig:
     """Wrapper around sshconf with some extra niceties."""
 
-    def __init__(self, path):
+    def __init__(self, path: str | Path):
```
Comment (JulesGM) [inline milatools/cli/utils.py:142]: this will constrain the python version maybe more than you want too

Comment (lebrice) [reply milatools/cli/utils.py:142]: There's a `from __future__ import annotations` at the top, so this only limits to Python 3.7+

Comment (lebrice) [reply milatools/cli/utils.py:142]: (https://peps.python.org/pep-0563/ for reference)

Comment (JulesGM) [reply milatools/cli/utils.py:142]: cool, didn't know about that import

**Review by breuleux — APPROVED** [review]

**Pull request merged**